index.js

```diff
@@ -1,158 +1,195 @@
+    }, (err, res, body) => {
+        try {
             if (err) throw 'Request Error: ' + err;
-            if (!body.sid) throw 'Login Error: SID is not defined.' + res;
-            sid = body.sid;
-        }).catch((err) => {
-            throw 'Error while calling Login: ' + err;
-        });
-        return sid;
-    },
+            body = JSON.parse(body);
+            body.communityList.forEach((element) => {
+                communityList.coms.push(sorter.comSort(element));
+            });
+            communityList.status = 'ok';
+            communityList.error = null;
+        } catch (err) {
+            communityList.error = err;
+        }
+    });
+    return communityList;
+}
+
+/**
+ * Loads all Kind of Chat Infomations that the Person itself joined.
+ * @param {SecurityString} sid For authenticating with the Narvii-API.
+ * @param {CommunityUUID} com A ID that can be obtained by the function getJoinedComs
+ * @returns {Object} Object where all the Chats that the Logged-in User has joined are contained in an Array.
+ */
+async function getJoinedChats(sid, com) {
+    let threadList = objs.threadList;
+    if(typeof sid != 'string' || typeof com !== 'string') {
+        throw new Error('All Arguments are not satisfied.');
+    }
+    await request.get(endpoints.getJoinedChats(com), {
+        headers: {
+            'NDCAUTH': `sid=${sid}`
+        }
+    }, (err, res, body) => {
+        try {
+            //Parsing the Response.
+            body = JSON.parse(body);
+            body.threadList.forEach((element) => {
+                //Sorting the Elements and pushing them into the Array.
+                threadList.threads.push(sorter.threadSort(element));
+            });
+            threadList.status = 'ok';
+            threadList.error = null;
+        } catch (err) {
+            threadList.error = err;
+        }
+    });
+    return threadList;
+}
 
-    getJoinedComs: async function(sid) {
-        let communityList = objs.communityList;
-        communityList.coms = [];
-        await request.get(endpoints.getComs, {
+/**
+ * Loads Messages in a specific Chat.
+ * @param {SecurityString} sid For authenticating with the Narvii-API.
+ * @param {CommunityUUID} com The Community ID that can be Obtained by the function getJoinedComs
+ * @param {ChatUUID} uid The Chats ID that can be obtained by the function getJoinedChats
+ * @param {Number} count The ammount of Messages to Load (defaults to 1);
+ * @returns {Object} Object where all the Messages in the requested Chat are contained in an Array.
+ */
+async function getChat(sid, com, uid, count) {
+    let msgList = objs.recivedMessages;
+    if(typeof sid != 'string' || typeof com !== 'string' || typeof uid !== 'string') {
+        throw new Error('All Arguments are not satisfied.');
+    }
+    if(count == undefined || count == null) {
+        count = 1;
+        console.log('[AminoJS.getChat()] No count Variable Defined. Defaulting to 1'); // eslint-disable-line 

```

Comment:
General Question. Should this Stay here?

Reply:
A silent fallback is more favorable.

Reply:
will do.